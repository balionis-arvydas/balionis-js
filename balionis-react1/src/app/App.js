import React from 'react';
import './App.css';

function App() {
  return (
    <h1 className="app">
        Hello, balionis-react1...
    </h1>
  );
}

export default App;
