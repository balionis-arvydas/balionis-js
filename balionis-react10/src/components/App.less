// @import "colors.less";
.appWrapper {
    color: yellow;
}
