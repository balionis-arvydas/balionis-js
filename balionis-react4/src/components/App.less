// @import "colors.less";
.App-wrapper {
    color: yellow;
}
