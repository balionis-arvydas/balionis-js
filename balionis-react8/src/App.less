.appWrapper {
    border: 1px solid #ff00ff;
}

.widget {
    border: 1px solid green;
}
