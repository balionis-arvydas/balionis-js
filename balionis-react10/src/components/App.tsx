import * as React from "react";
import { Tabs, Tab, TabId } from "@blueprintjs/core";

import "./App.less";

type AppProps = {}

type AppState = {
    selectedTabId: TabId
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);

        this.state = {
            selectedTabId: "0"
        }
    }

    handleTabChange = (selectedTabId: TabId) => {
        this.setState({
            selectedTabId
        });
    }

    render() {
        return (
            <div className="appWrapper">
                <Tabs id="TabsExample" onChange={this.handleTabChange} selectedTabId={this.state.selectedTabId}>
                    <Tab id="0" title="Angular" />
                    <Tab id="1" title="Ember" panelClassName="ember-panel" />
                    <Tab id="2" title="React" />
                    <Tab id="3" disabled title="Backbone" />
                    <Tabs.Expander />
                </Tabs>
            </div>
        );
    }
}

export default App;
