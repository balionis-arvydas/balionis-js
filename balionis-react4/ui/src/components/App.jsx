import React, { Component } from "react";
import "./App.less";

class App extends Component {
  constructor() {
    super();

    this.state = {
      value: "Hello World."
    };
  }
  
  render() {
    return (
        <p className="App-wrapper">{this.state.value}</p>
    );
  }
}  

export default App;
