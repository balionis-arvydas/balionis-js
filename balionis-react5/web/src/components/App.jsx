import React, { Component } from "react";
import "./App.less";
import { readUser } from "../actions/userActions";

class App extends Component {
    constructor () {
        super();
        this.state = { value: "", disabled: true, responses: [] };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange (event) {
        const value = event.target.value;
        const disabled = value === null || value.length < 3;
        this.setState({
            value,
            disabled
        });
    }

    handleSubmit (event) {
        event.preventDefault();
        readUser(this.state.value)
            .then(data => {
                const responses = [...this.state.responses, data];
                this.setState({
                    value: '',
                    disabled: false,
                    responses,
                });
            });
    }

    render () {
        const responses = this.state.responses.map((x, i) => {
            return (
                <div className="message" key={i}>{x}</div>
            );
        });

        return (
            <div className="App-wrapper">
                <div className="adder">
                    <input type="text" value={this.state.value} onChange={this.handleChange} />
                    <button onClick={this.handleSubmit} disabled={this.state.disabled}>Add</button>
                </div>
                <div className="responses">
                    {responses}
                </div>
            </div>
        );
    }
}

export default App;
