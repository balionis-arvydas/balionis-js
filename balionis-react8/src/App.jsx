import React from "react";

import RGL, { WidthProvider as widthProvider } from "react-grid-layout";

import "react-grid-layout/css/styles.css";
import "react-resizable/css/styles.css";
import "./App.less";

import WidgetFactory from "./components/widgetFactory";

const ReactGridLayout = widthProvider(RGL);

class App extends React.Component {
    constructor (props) {
        super(props);

        const { layout, widgets } = this.buildWorkspace();
        this.state = { layout, widgets };
    }

    buildWorkspace () {
        return {
            layout: [
                { i: '0', x: 0, y: 0, w: 20, h: 2, static: true },
                { i: '1', x: 0, y: 2, w: 20, h: 5 },
                { i: '2', x: 0, y: 10, w: 20, h: 10 }
            ],
            widgets: [
                { widgetType: "AddTask" },
                { widgetType: "TaskList" },
                { widgetType: "TaskList" }
            ]
        };
    }

    renderWidgets () {
        return this.state.widgets.map((widget, i) => {
            const component = WidgetFactory.buildWidget(widget.widgetType, this.props);
            return (
                <div className="widget" key={i}>
                    {component}
                </div>);
        });
    }

    onLayoutChange (layout) {
        console.log("onLayoutChange: layout=", layout);
    }

    render () {
        return (
            <div data-automation-id="app" className="appWrapper">
                <ReactGridLayout
                    className="layout" cols={60} rowHeight={10} width={1200}
                    layout={this.state.layout}
                    onLayoutChange={this.onLayoutChange}
                    {...this.props}
                >
                    {this.renderWidgets()}
                </ReactGridLayout>
            </div>
        );
    }
}

export default App;
