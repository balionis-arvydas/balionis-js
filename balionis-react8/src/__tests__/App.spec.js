import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mockStore } from '../redux/__tests__/mockStore';

import App from '../App';

describe('AppComponent', () => {
    let store;

    beforeAll(() => {
        jest.resetAllMocks();
    });

    beforeEach(() => {
        store = mockStore({
            taskList: { tasks: [] }
        });
        store.dispatch = jest.fn();
    });

    test("snapshot", async () => {
        const component = renderer.create(
            <Provider store={store}>
                <App/>,
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

});
