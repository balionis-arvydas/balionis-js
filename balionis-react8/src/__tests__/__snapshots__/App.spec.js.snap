// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapshot 1`] = `
Array [
  <div
    className="appWrapper"
    data-automation-id="app"
  >
    <div
      className="react-grid-layout layout"
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      style={
        Object {
          "height": "350px",
        }
      }
    >
      <div
        className="react-grid-item widget static react-resizable-hide react-resizable"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        style={
          Object {
            "height": "30px",
            "left": "0.78125%",
            "position": "absolute",
            "top": "10px",
            "width": "32.265625%",
          }
        }
      >
        <div
          className="addTaskWrapper"
        >
          <input
            className="addTaskInput"
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            className="addTaskButton"
            disabled={true}
            onClick={[Function]}
          >
            Add
          </button>
        </div>
        <span
          className="react-resizable-handle react-resizable-handle-se"
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
        />
      </div>
      <div
        className="react-grid-item widget react-draggable react-resizable"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        style={
          Object {
            "height": "90px",
            "left": "0.78125%",
            "position": "absolute",
            "top": "50px",
            "width": "32.265625%",
          }
        }
      >
        <div
          className="taskListWrapper"
        />
        <span
          className="react-resizable-handle react-resizable-handle-se"
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
        />
      </div>
      <div
        className="react-grid-item widget react-draggable react-resizable"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        style={
          Object {
            "height": "190px",
            "left": "0.78125%",
            "position": "absolute",
            "top": "150px",
            "width": "32.265625%",
          }
        }
      >
        <div
          className="taskListWrapper"
        />
        <span
          className="react-resizable-handle react-resizable-handle-se"
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
        />
      </div>
    </div>
  </div>,
  ",",
]
`;
